engine/src/flutter/shell/platform/darwin/ios/framework/Source/FlutterTextInputPluginTest.mm

```diff
@@ -2987,6 +2990,264 @@ - (void)testEditMenu_shouldPresentEditMenuWithCorectTargetRect {
   }
 }
 
+- (void)testEditMenu_shouldPresentEditMenuWithSuggestedItemsByDefaultIfNoFrameworkData {
+  if (@available(iOS 16.0, *)) {
+    FlutterTextInputPlugin* myInputPlugin =
+        [[FlutterTextInputPlugin alloc] initWithDelegate:OCMClassMock([FlutterEngine class])];
+    FlutterViewController* myViewController = [[FlutterViewController alloc] init];
+    myInputPlugin.viewController = myViewController;
+    [myViewController loadView];
+
+    FlutterMethodCall* setClientCall =
+        [FlutterMethodCall methodCallWithMethodName:@"TextInput.setClient"
+                                          arguments:@[ @(123), self.mutableTemplateCopy ]];
+    [myInputPlugin handleMethodCall:setClientCall
+                             result:^(id _Nullable result){
+                             }];
+
+    FlutterTextInputView* myInputView = myInputPlugin.activeView;
+
+    FlutterTextInputView* mockInputView = OCMPartialMock(myInputView);
+    OCMStub([mockInputView isFirstResponder]).andReturn(YES);
+
+    XCTestExpectation* expectation = [[XCTestExpectation alloc]
+        initWithDescription:@"presentEditMenuWithConfiguration must be called."];
+
+    id mockInteraction = OCMClassMock([UIEditMenuInteraction class]);
+    OCMStub([mockInputView editMenuInteraction]).andReturn(mockInteraction);
+    OCMStub([mockInteraction presentEditMenuWithConfiguration:[OCMArg any]])
+        .andDo(^(NSInvocation* invocation) {
+          [expectation fulfill];
+        });
+
+    myInputView.frame = CGRectMake(10, 20, 30, 40);
+    NSDictionary<NSString*, NSNumber*>* encodedTargetRect =
+        @{@"x" : @(100), @"y" : @(200), @"width" : @(300), @"height" : @(400)};
+    // No items provided from framework. Show the suggested items by default.
```

Comment:
We discussed offline a bit and I summarized in https://github.com/flutter/flutter/pull/159013#issuecomment-2634939771 that `items` should be required, and since that would be a breaking change, we have to support both use cases for now while I deprecate the old API.

Are there any changes you want to make in this PR to prepare for that future? Is it possible to add deprecation notices in the engine? Or at least here and other places where it comes up, explain in a comment.

Reply:
Since nothing has changed about the framework <-> engine communication, I think we are good for now. The deprecation happens in dart API only.

Reply:
Sounds good. When we do remove the deprecated `show` method, we should also tighten up the engine API to require `items` to be present and not empty.